sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -170,6 +174,125 @@ class JDBCTableCatalog extends TableCatalog with Logging {
     }
   }
 
+  override def namespaceExists(namespace: Array[String]): Boolean = namespace match {
+    case Array(db) =>
+      listNamespaces.exists (_(0) == db)
+    case _ => false
+  }
+
+  override def listNamespaces(): Array[Array[String]] = {
+    withConnection { conn =>
+      val catalogs = conn.getMetaData.getCatalogs()
```

Comment:
does JDBC v2 connect to a certain catalog?

Reply:
Sorry for the late reply. I was OoO last week.

Yes, it connects to the current database, which is specified in the connection url.
For example: 
```postgres``` in ```jdbc:postgresql://127.0.0.1:5432/postgres;user=testUser;password=testPass```
 or ```testdb0``` in ```jdbc:h2:mem:testdb0;user=testUser;password=testPass```